include/accessor.h

```diff
@@ -477,6 +489,51 @@ namespace detail {
 		}
 	}
 
+
+	template <typename DT, cl::sycl::access::mode M, cl::sycl::target T>
+	class accessor_subscript_proxy<DT, 3, M, T, 0> {
+		using AccessorT = celerity::accessor<DT, 3, M, T>;
+
+	  public:
+		accessor_subscript_proxy(const AccessorT& acc, const size_t d0, const size_t d1) : acc(acc), d0(d0), d1(d1) {}
+
+		DT& operator[](const size_t dx) const { return acc[{d0, d1, dx}]; }
+
+	  private:
+		const AccessorT& acc;
+		size_t d0;
+		size_t d1;
+	};
+
+	template <typename DT, cl::sycl::access::mode M, cl::sycl::target T>
+	class accessor_subscript_proxy<DT, 3, M, T, 1> {
+		using AccessorT = celerity::accessor<DT, 3, M, T>;
+
+	  public:
+		accessor_subscript_proxy(const AccessorT& acc, const size_t d0) : acc(acc), d0(d0) {}
+
+		accessor_subscript_proxy<DT, 3, M, T, 0> operator[](const size_t dx) const { return accessor_subscript_proxy<DT, 3, M, T, 0>{acc, d0, dx}; }
```

Comment:
As mentioned in response to @fknorr's comment, this or the specialization below is redundant and can be combined into a single specialization with `decltype(auto)` return type and `constexpr if` (or SFINAE, because of that CCPP compiler bug... in any case, we could get rid of a few lines of duplicated code).